uniffi_bindgen/src/interface/mod.rs

```diff
@@ -126,9 +128,19 @@ impl ComponentInterface {
         ci.types.add_type_definitions_from(defns.as_slice())?;
         // With those names resolved, we can build a complete representation of the API.
         APIBuilder::process(&defns, &mut ci)?;
+
+        // The FFI namespace must not be computed on the fly because it could otherwise be
+        // influenced by things added later from proc-macro metadata. Those have their own
+        // namespacing mechanism.
+        assert!(!ci.namespace.is_empty());
+        ci.ffi_namespace = format!("{}_{:x}", ci.namespace, ci.checksum());
```

Comment:
> I think this could be done in a separate PR

Definitely. Doesn't really matter whether it would be merged before or after this PR, right?

Reply:
After thinking about this more, the issue with dropping the interface-wide namespace is trying to combine multiple UniFFI crates in the same .so, like we currently do.  In that case, you need each `rustbuffer_alloc` function to have a different name (for now at least, at some point maybe we could figure out a way to avoid this duplication).  Also, if 2 functions from different UniFFIed crates have the same name and signature then we would get a collision.

It's unfortunate that `derive_ffi_funcs()` needs to stay and the way we generate the prefix differs between the UDL approach and the macro-based approach, but I think we can live with that.